### DISCUSSION TITLE
Shelly 2PM Gen3

### DISCUSSION THREAD

**Discussion by Judocrafter** [mongoose-os-apps/shelly-homekit#1538] [Ideas]
Will the Shelly 2PM Gen3 also be supported with this Homekit software in the future?

**Comment by markirb** [comment]
Sure there is no reason why not. As soon as we know the Pinning and the Powerchip, this should be easy

**Reply by Judocrafter** [reply]
Great! Can hardly wait. How long do you think it will take? Will it then also be possible to control the tilt function of roller shutters in Homekit?

**Reply by markirb** [reply]
Yes it will be possible. If I have the information then several days. So far I do have nothing.

**Reply by MohdAtieh** [reply]
Hey, Any updates on the support of the 2PMGen3

**Reply by Judocrafter** [reply]
> Hey, Any updates on the support of the 2PMGen3

I haven't heard anything yet. However, as a workaround, I have integrated the Shelly 2pm Gen3 into Homeassistant and from there into Homekit via HomeKit Bridge.
Works perfectly, only the blinds control is a bit special, but this is probably also due to Homekit itself.

**Reply by MohdAtieh** [reply]
Gen 2 or 3 you mean ??

**Reply by thepilzkopf** [reply]
> Works perfectly, only the blinds control is a bit special, but this is probably also due to Homekit itself.

What do you mean with special? Planning to use it this way by end of the week until this firmware here is ready for the 2PM Gen3.

**Reply by Judocrafter** [reply]
> Gen 2 or 3 you mean ??

sorry I mean gen3 of course, have now adjusted it above.

**Reply by Judocrafter** [reply]
> > Works perfectly, only the blinds control is a bit special, but this is probably also due to Homekit itself.
> 
> What do you mean with special? Planning to use it this way by end of the week until this firmware here is ready for the 2PM Gen3.

It is a bit special to control the tilt angle via homekit, as this is not solved as nicely on the “main side” as in shelly. Do you understand what I mean? But it works wonderfully and I don't currently need the tilt angle.
![IMG_3955](https://github.com/user-attachments/assets/29831146-2ff9-4e19-bb07-6c5b04fb6c4a)
![IMG_3956](https://github.com/user-attachments/assets/1e6ebe25-a180-40a5-b066-a6161a4349dd)
![IMG_3957](https://github.com/user-attachments/assets/b721acd6-d3bf-49c5-8be4-0b2b3eea989c)

**Reply by thepilzkopf** [reply]
Looks nice, thanks! I understand what you mean

**Reply by thepilzkopf** [reply]
> Sure there is no reason why not. As soon as we know the Pinning and the Powerchip, this should be easy

Is there a possibility to help you with this?

**Reply by markirb** [reply]
> > Sure there is no reason why not. As soon as we know the Pinning and the Powerchip, this should be easy
> 
> Is there a possibility to help you with this?

If you have electronics background you could measure :) if you have not you can ask Shelly support

**Reply by patrickweegen** [reply]
Is this feature in the current version?
I've just installed some 2PMG3 and I can not control the tilt of the blinds

**Comment by markirb** [comment]
There is a preliminary pinning from tasmota discussions, I have put it into a PR.

If anyone feels adventurous:

https://github.com/mongoose-os-apps/shelly-homekit/pull/1578/checks

**Reply by markirb** [reply]
Anyone wants to try and give feeeback?

**Reply by Judocrafter** [reply]
> Anyone wants to try and give feeeback?

I would be very happy to try it out. Can you briefly explain how I can help you?

**Reply by markirb** [reply]
[shelly-homekit-Shelly2PMGen3.zip](https://github.com/user-attachments/files/18038525/shelly-homekit-Shelly2PMGen3.zip)

try to flash this (right click save as )

**Reply by Judocrafter** [reply]
> [shelly-homekit-Shelly2PMGen3.zip](https://github.com/user-attachments/files/18038525/shelly-homekit-Shelly2PMGen3.zip)
> 
> try to flash this (right click save as )

thank you, i'll try to install it this weekend

**Reply by Judocrafter** [reply]
> [shelly-homekit-Shelly2PMGen3.zip](https://github.com/user-attachments/files/18038525/shelly-homekit-Shelly2PMGen3.zip)
> 
> try to flash this (right click save as )

unfortunately it didn't work at all. overheating. do you have the link to return to the normal version?

**Reply by markirb** [reply]
Don't let this scare yourself, It is not really overheating...
Only showing wrong temperature.

I would not call that not working at all I would call that a part success.

The Temperature should be fixed here, I saw what I did wrong in the first place:
[shelly-homekit-Shelly2PMGen3.zip](https://github.com/user-attachments/files/18042111/shelly-homekit-Shelly2PMGen3.zip)

**Reply by markirb** [reply]
tasmota template got updated several times.
with this one there's also the reset for the power ic, so it could work also

[shelly-homekit-Shelly2PMGen3.zip](https://github.com/user-attachments/files/18042419/shelly-homekit-Shelly2PMGen3.zip)

**Reply by Judocrafter** [reply]
@markirb i was able to flash the firmware and add the shelly to homekit. however, it then loses the connection as soon as i start the calibration. is it possible to revert to the original firmware? i am away until sunday and don't want to take the risk of the shelly burning out

![image](https://github.com/user-attachments/assets/1a028417-f1ea-413b-907b-509b738c21a7)
![Screenshot 2024-12-06 200748](https://github.com/user-attachments/assets/98a12269-88fd-4087-b036-97267b8ad6e8)

**Reply by Judocrafter** [reply]
ah and if i want to search for updates here it also failed. but i assume this is still the case in this version
![image](https://github.com/user-attachments/assets/9c621aa7-5154-4733-badb-d1e84af4285c)

**Reply by markirb** [reply]
Ah, so you are using window opening... bad. Because I figure everything should work fine except for powermeter, as the whole i2c port is new for esp32c3.

URLs only work once it is out of this pre version.

Unfortunately there is nothing to revert to at least the usual link does not provide a firmware to stock yet:

https://updates.shelly.cloud/update/S2PMG3

Can you test in non window mode?

**Reply by Judocrafter** [reply]
@markirb 
yes I use it for a roller shutter which I also wrote above? So I can only test it in this function. So I have no possibility to return to the normal version?
But perhaps you could have mentioned that briefly...
Can I leave the Shelly attached over the weekend without any problems?

**Reply by markirb** [reply]
Yes but still it can be operated on normal mode, maybe we see better what's wrong there. One output controls each direction then.

I'm sorry. It was mentioned on issues. 

What is your concern? If nobody touches the device then the device does nothing.

**Reply by Judocrafter** [reply]
okay we'll just leave tomorrow morning and no one will press the switch. i'll test it at the beginning of the week with the
normal mode okay? i just wanted to be sure that nothing starts to burn haha

**Reply by markirb** [reply]
a beta stock firmware is now available to revert:

https://updates.shelly.cloud/update/S2PMG3
